extensions/amp-script/0.1/amp-script.js

```diff
@@ -117,19 +118,14 @@ export class AmpScript extends AMP.BaseElement {
      *   2. The amp-script tag or any of its parents must also have 'data-ampdevmode'.
      */
     const devTag = 'data-ampdevmode';
-    let parentHasDevTag = false;
-    if (this.element.ownerDocument.documentElement.hasAttribute(devTag)) {
-      let parentWalker = this.element;
-      while (parentWalker != null) {
-        if (parentWalker.hasAttribute('data-ampdevmode')) {
-          parentHasDevTag = true;
-          break;
-        }
-        parentWalker = parentWalker.parentNode;
-      }
-    }
+    const doc = this.element.ownerDocument.documentElement;
+    const closestAncestorWithDevTag = closestAncestorElementBySelector(
+      this.element,
+      `[${devTag}]:not(html)`
```

Comment:
Nit: We can just check that the returned element is not `html` instead of doing a more expensive selector here.

Reply:
Thats actually how I'd done it originally. Is there anywhere you can point me to so I can learn which selectors are/are not expensive?

Reply:
Not really. Basically, `*` is super-ultra-mega expensive, and combining multiple selectors (or multiple conditions on a single selector) is more up to 2x more expensive.